pkgs/development/tools/nailgun/default.nix

```diff
@@ -8,33 +8,56 @@ let
     inherit version;
     sha256 = "1mk8pv0g2xg9m0gsb96plbh6mc24xrlyrmnqac5mlbl4637l4q95";
   };
-in
-stdenv.mkDerivation {
-  pname = "nailgun";
-  inherit version;
 
-  src = fetchFromGitHub {
-    owner = "facebook";
-    repo = "nailgun";
-    rev = "nailgun-all-v${version}";
-    sha256 = "1syyk4ss5vq1zf0ma00svn56lal53ffpikgqgzngzbwyksnfdlh6";
-  };
+  license = lib.licenses.asl20;
+  homepage = "http://www.martiansoftware.com/nailgun/";
+  platforms = lib.platforms.linux;
+  maintainers = with lib.maintainers; [ ];
 
-  makeFlags = [ "PREFIX=$(out)" ];
+  server = runCommand "nailgun-server-${version}" {
+    pname = "nailgun-server";
+    inherit version;
 
-  nativeBuildInputs = [ makeWrapper ];
+    nativeBuildInputs = [ makeWrapper ];
 
-  postInstall = ''
+    meta = {
+      inherit license homepage platforms maintainers;
+      description = "Server for running Java programs from the command line without incurring the JVM startup overhead";
+      sourceProvenance = with lib.sourceTypes; [ binaryBytecode ];
+    };
+  } ''
     makeWrapper ${jre}/bin/java $out/bin/ng-server \
       --add-flags '-classpath ${nailgun-server.jar}:$CLASSPATH com.facebook.nailgun.NGServer'
   '';
 
-  meta = with lib; {
+  client = stdenv.mkDerivation {
+    pname = "nailgun-client";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "facebook";
+      repo = "nailgun";
+      rev = "nailgun-all-v${version}";
+      sha256 = "1syyk4ss5vq1zf0ma00svn56lal53ffpikgqgzngzbwyksnfdlh6";
+    };
+
+    makeFlags = [ "PREFIX=$(out)" ];
+
+    meta = {
+      inherit license homepage platforms maintainers;
+      description = "Client for running Java programs from the command line without incurring the JVM startup overhead";
+    };
+  };
+in
+symlinkJoin rec {
+  pname = "nailgun";
+  inherit client server version;
```

Comment:
Why do we need a package that has both? You can just install both packages which should both be exposed in top-level. We can remove this.

Reply:
Backward compatibility? I think wanting to install only one or the other of these is an edge case, and I'm not convinced it'd be worth disrupting the configs of most users of nailgun to achieve this.

Reply:
The package is only a few days old. It's fine to break compatibility here.

Reply:
... What? It was [added in 2017](https://github.com/NixOS/nixpkgs/commit/649a036dccfd2f23179582c3b995046833bd0b26).